File: Samples/Verbum/associate-an-application-with-a-particular-file-extension-to-open-it-by-this-application-e3207-13.1.4-/CS/FileAssociationHelper.cs
using Microsoft.Win32;
using System.Windows.Forms;
using Microsoft.VisualBasic.Devices;

public static class FileAssociationHelper {
    public static readonly string DefaultRegistryKeyValueName = string.Empty;

    public static void AssociateFileExtension(string fileExtension) {
        string actualFileExtension = (fileExtension[0] == '.' ? fileExtension : '.' + fileExtension);
        string friendlyName = actualFileExtension.TrimStart(new char[] { '.' }) + "file";

        Computer computer = new Computer();

        RegistryKey rkFileExtension = computer.Registry.ClassesRoot.CreateSubKey(actualFileExtension);
        rkFileExtension.SetValue(DefaultRegistryKeyValueName, friendlyName, RegistryValueKind.String);

        RegistryKey rkShellOpenCommand = computer.Registry.ClassesRoot.CreateSubKey(friendlyName + @"\shell\open\command");
        rkShellOpenCommand.SetValue(DefaultRegistryKeyValueName, Application.ExecutablePath + @" ""%l"" ", RegistryValueKind.String);

        RegistryKey rkDefaultIcon = computer.Registry.ClassesRoot.CreateSubKey(friendlyName + @"\DefaultIcon");
        rkDefaultIcon.SetValue(DefaultRegistryKeyValueName, Application.StartupPath + @"\App.ico", RegistryValueKind.String);
    }
}

File: Samples/Verbum/associate-an-application-with-a-particular-file-extension-to-open-it-by-this-application-e3207-13.1.4-/VB/FileAssociationHelper.vb
Imports Microsoft.Win32
Imports System.Windows.Forms
Imports Microsoft.VisualBasic.Devices

Public NotInheritable Class FileAssociationHelper
	Public Shared ReadOnly DefaultRegistryKeyValueName As String = String.Empty

	Private Sub New()
	End Sub
	Public Shared Sub AssociateFileExtension(ByVal fileExtension As String)
		Dim actualFileExtension As String
		If fileExtension.Chars(0) = "."c Then
			actualFileExtension = (fileExtension)
		Else
			actualFileExtension = ("."c & fileExtension)
		End If
		Dim friendlyName As String = actualFileExtension.TrimStart(New Char() { "."c }) & "file"

		Dim computer As New Computer()

		Dim rkFileExtension As RegistryKey = computer.Registry.ClassesRoot.CreateSubKey(actualFileExtension)
		rkFileExtension.SetValue(DefaultRegistryKeyValueName, friendlyName, RegistryValueKind.String)

		Dim rkShellOpenCommand As RegistryKey = computer.Registry.ClassesRoot.CreateSubKey(friendlyName & "\shell\open\command")
		rkShellOpenCommand.SetValue(DefaultRegistryKeyValueName, Application.ExecutablePath & " ""%l"" ", RegistryValueKind.String)

		Dim rkDefaultIcon As RegistryKey = computer.Registry.ClassesRoot.CreateSubKey(friendlyName & "\DefaultIcon")
		rkDefaultIcon.SetValue(DefaultRegistryKeyValueName, Application.StartupPath & "\App.ico", RegistryValueKind.String)
	End Sub
End Class